### PR TITLE
added jpeg qr functionality using a modified version of jpec

### PR DESCRIPTION
**Pull request by michilumin**
now writes jpegs too because moms and dads love jpeg

### CODE REVIEW

File: src/qrengine.c [hunk 1]
```diff
@@ -185,4 +218,47 @@ int qrgen_string_to_qr_pngfile(const char *filename, const char* inString, uint8
...
+        fclose(file);
+        jpec_enc_del(enc);
+        free(image);
+    } 
+    else 
+    {
+        printf("Error generating QR\n");
+        return 1;
+    }
 }
```
Comment (xanimo) [inline src/qrengine.c:264]: New line. Also perhaps consider as 1 is truthy and the enclosure of codeGenerated doesn't return anything:
```

// encode a string to JPEG file with med ECC
int qrgen_string_to_qr_jpgfile(const char* outFilename, const char* inString, uint8_t sizeMultiplier)
{
    if (sizeMultiplier < 1) {
        sizeMultiplier = 1;
    }

    // unsigned char* jpg;
    int jpgsize;
    unsigned width;
    unsigned height;

    uint8_t tempBuffer[qrcodegen_BUFFER_LEN_MAX];
    uint8_t qrcode[qrcodegen_BUFFER_LEN_MAX];

    bool codeGenerated = qrcodegen_encodeText(inString, tempBuffer, qrcode, qrcodegen_Ecc_MEDIUM, qrcodegen_VERSION_MIN, qrcodegen_VERSION_MAX, qrcodegen_Mask_AUTO, true);

    if (codeGenerated) {
        unsigned tempSideDim = qrcodegen_getSize(qrcode);
        width = tempSideDim * sizeMultiplier;
        height = tempSideDim * sizeMultiplier;

        // bytes to monobitmap with sizemodifier
        uint8_t* image = bytesToMono(qrcode, sizeMultiplier);

        //encode jpeg
        jpec_enc_t* enc = jpec_enc_new(image, width, height);
        const uint8_t* jpeg = jpec_enc_run(enc, &jpgsize);

        FILE* file = fopen(outFilename, "wb");
        fwrite(jpeg, sizeof(uint8_t), jpgsize, file);
        fclose(file);
        jpec_enc_del(enc);
        free(image);
    } 
    else 
    {
        printf("Error generating QR\n");
        return 0;
    }
    return codeGenerated;
}

```

**Review by xanimo — COMMENTED** [review]
I'd add qrtest.jpg to .gitignore and perhaps consider refactoring that codeGenerated section so it will return something on either conditional outcome. Outside of that tested on nix and it's working great.

File: src/qrengine.c (hunk 1 above)
Comment (michilumin) [reply src/qrengine.c]: think i got it all now.

**Review by xanimo — APPROVED** [review]
ACK

**Pull request merged**